BaseClasses.py

```diff
@@ -427,6 +429,15 @@ def get_entrance(self, entrance_name: str, player: int) -> Entrance:
     def get_location(self, location_name: str, player: int) -> Location:
         return self.regions.location_cache[player][location_name]
 
+    def get_location_if_available(self, location_name: str, player: int) -> Optional[Location]:
+        """Like get_location, but returns None if the location isn't in the world.
```

Comment:
New docstrings should follow full reST style https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/style.md?plain=1#L29